unison-cli/src/Unison/CommandLine/Main.hs

```diff
@@ -116,10 +117,12 @@ main dir welcome initialPath config initialInputs runtime sbRuntime codebase ser
       -- Try putting the root, but if someone else as already written over the root, don't
       -- overwrite it.
       void $ tryPutTMVar rootVar root
-    -- Start forcing the thunk in a background thread.
+    -- Start forcing thunks in a background thread.
     -- This might be overly aggressive, maybe we should just evaluate the top level but avoid
     -- recursive "deep*" things.
-    void $ UnliftIO.evaluate root
+    UnliftIO.concurrently_
+      (UnliftIO.evaluate root)
+      (UnliftIO.evaluate typecheckedFile) -- IOSource takes a while to compile, we should start compiling it on startup
```

Comment:
Potentially we can remove this if it compiles on-demand fast enough, but it's also probably fine to leave here.